src/sentry/shared_integrations/client/base.py

```diff
@@ -329,3 +378,66 @@ def get_with_pagination(
             if num_results < page_size:
                 return output
         return output
+
+    def record_response(self, response: BaseApiResponse):
+        if self.is_response_fatal(response):
+            self.record_request_fatal(response)
+        elif self.is_response_error(response):
+            self.record_request_error(response)
+        elif self.is_response_success(response):
+            self.record_request_success(response)
+
+    def record_error(self, error: Exception):
+        redis_key = self._get_redis_key()
+        if not len(redis_key):
```

Comment:
@chloeho7 when would this happen?

Reply:
Sometimes `SlackClient` are made that don't have integration id for example in `_get_team_info` in `sentry/src/sentry/integrations/slack/integration.py`